pyInterface/package/_likelihood.py

```diff
@@ -1,65 +1,93 @@
+ 
+ 	normIntFile = ROOT.TFile.Open(normIntegralFileName, "READ")
+ 	if len(normIntFile.GetListOfKeys()) != 1:
+ 		pyRootPwa.utils.printWarn("'" + normIntegralFileName + "' does not contain exactly one TKey.")
 		return None
-	normIntMatrix = normIntFile.Get(pyRootPwa.core.ampIntegralMatrix.integralObjectName)
-	if (not likelihood.addNormIntegral(normIntMatrix)):
-		pyRootPwa.utils.printErr("could not add normalization integral. Aborting...")
+ 	normIntMatrix = normIntFile.Get(pyRootPwa.core.ampIntegralMatrix.integralObjectName)
+ 	if (not likelihood.addNormIntegral(normIntMatrix)):
+ 		pyRootPwa.utils.printErr("could not add normalization integral. Aborting...")
 		return None
-	normIntFile.Close()
-	accIntFile = ROOT.TFile.Open(accIntegralFileName, "READ")
-	if len(accIntFile.GetListOfKeys()) != 1:
-		pyRootPwa.utils.printWarn("'" + normIntegralFileName + "' does not contain exactly one TKey.")
+ 	normIntFile.Close()
+ 	accIntFile = ROOT.TFile.Open(accIntegralFileName, "READ")
+ 	if len(accIntFile.GetListOfKeys()) != 1:
+ 		pyRootPwa.utils.printWarn("'" + normIntegralFileName + "' does not contain exactly one TKey.")
 		return None
-	accIntMatrix = accIntFile.Get(pyRootPwa.core.ampIntegralMatrix.integralObjectName)
-	if (not likelihood.addAccIntegral(accIntMatrix, accEventsOverride)):
-		pyRootPwa.utils.printErr("could not add acceptance integral. Aborting...")
+ 	accIntMatrix = accIntFile.Get(pyRootPwa.core.ampIntegralMatrix.integralObjectName)
+ 	if (not likelihood.addAccIntegral(accIntMatrix, accEventsOverride)):
+ 		pyRootPwa.utils.printErr("could not add acceptance integral. Aborting...")
 		return None
-	accIntFile.Close()
-
-	for wave in waveDescThres:
-		waveName = wave[0]
-		ampFileName = ampFileList[waveName]
-		ampFile = ROOT.TFile.Open(ampFileName, "READ")
-		if not ampFile:
-			pyRootPwa.utils.printErr("could not open amplitude file '" + ampFileName + "'.")
+ 	accIntFile.Close()
+	eventMetas = []
+	openFiles = []
+	for evtFileName in eventFileNameList:
+		evtFile = ROOT.TFile.Open(evtFileName, "READ")
+		if not evtFile:
+			pyRootPwa.utils.printErr("could not open amplitude file '" + evtFileName + "'.")
 			return None
-		meta = pyRootPwa.core.amplitudeMetadata.readAmplitudeFile(ampFile, waveName)
-		if not meta:
-			pyRootPwa.utils.printErr("could not get metadata for waveName '" + waveName + "'.")
+		evtMeta = pyRootPwa.core.eventMetadata.readEventFile(evtFile)
+		if not evtMeta:
+			pyRootPwa.utils.printErr("could not get metadata for event file '" + evtFileName + "'.")
 			return None
-		if (not likelihood.addAmplitude(meta)):
-			pyRootPwa.utils.printErr("could not add amplitude '" + waveName + "'. Aborting...")
+		eventMetas.append(evtMeta)
+		openFiles.append(evtFile)
+	likelihood.setOnTheFlyBinning(binningMap, eventMetas)
+ 	for wave in waveDescThres:
+		openFilesWave = []
+ 		waveName = wave[0]
+		ampFileNameList = ampFileList[waveName]
+		if not len(ampFileNameList) == len(eventMetas) and len(eventMetas) > 0:
```

Comment:
Other way around makes more sense, i.e.:

`if len(eventMetas) > 0 and not len(ampFileNameList) == len(eventMetas):`